test/test_torch.py

```diff
@@ -1232,7 +1232,7 @@ def test_masked_scatter(self):
                         src = torch.randn(num_copy - 1)
                         with self.assertRaises(RuntimeError):
                             dest.masked_scatter_(mask, src)
-            self.assertEqual(len(w), 27)
+            self.assertEqual(len(w), 20)
```

Comment:
Yeah, cuda variant is not tested here. Why did the number of warnings change?

Reply:
This happens because I think the test has bug.

For the snippet below,
`src` is always is of `dtype=torch.float`, while dtype of `dest` varies as per the loop.
```python
# make src smaller. this should fail
src = torch.randn(num_copy - 1)
with self.assertRaises(RuntimeError):
    dest.masked_scatter_(mask, src)
```

So except for `dest` and `src` having dtype `torch.float32`, all other cases fail with Error due to mismatch in dtypes of `dest` and `src`.

In the new implementation, we first check if the types are `src` and `dest` are same and proceed with the warning only then. However, in the old implementation, we check warning condition first and then check for types of `src` and `dest`.

That is why the discrepancy.
https://github.com/pytorch/pytorch/blob/cc003e69df3d231f852281f065b5a22dbacde0d0/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L1124-L1141

Actually will fix the bug.
Thanks!